File: app/src/main/java/com/zuk0/gaijinsmash/riderz/data/remote/retrofit/ApiKeyInterceptor.java
package com.zuk0.gaijinsmash.riderz.data.remote.retrofit;

import java.io.IOException;

import androidx.annotation.NonNull;
import okhttp3.HttpUrl;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class ApiKeyInterceptor implements Interceptor {

    private String mApiKey;
    private String mName;

    public ApiKeyInterceptor(String name, String apiKey) {
        mName = name;
        mApiKey = apiKey;
    }

    @Override
    public Response intercept(@NonNull Interceptor.Chain chain) throws IOException {
        HttpUrl url = chain.request().url()
                .newBuilder()
                .addQueryParameter(mName, mApiKey)
                .build();
        Request request = chain.request().newBuilder().url(url).build();
        return chain.proceed(request);
    }
}

File: app/src/main/java/com/zuk0/gaijinsmash/riderz/data/remote/retrofit/ApiKeyInterceptor.kt
package com.zuk0.gaijinsmash.riderz.data.remote.retrofit

import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

class ApiKeyInterceptor(private val mName: String, private val mApiKey: String) : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val url = chain.request().url
                .newBuilder()
                .addQueryParameter(mName, mApiKey)
                .build()
        val request = chain.request().newBuilder().url(url).build()
        return chain.proceed(request)
    }

}